x-pack/legacy/plugins/apm/server/routes/services.ts

```diff
@@ -86,15 +86,48 @@ export const serviceNodeMetadataRoute = createRoute(() => ({
   }
 }));
 
+export const serviceMapAllRoute = createRoute(() => ({
+  path: '/api/apm/service-map/all',
+  params: {
+    query: t.intersection([
+      t.partial({ environment: t.string }),
+      uiFiltersRt,
+      rangeRt
+    ])
+  },
+  handler: async ({ context, request }) => {
+    if (!context.config['xpack.apm.serviceMapEnabled']) {
+      return new Boom('Not found', { statusCode: 404 });
+    }
+    const setup = await setupRequest(context, request);
+    const {
+      query: { environment }
+    } = context.params;
+    return getServiceMap({ setup, environment });
+  }
+}));
+
 export const serviceMapRoute = createRoute(() => ({
-  path: '/api/apm/service-map',
+  path: '/api/apm/service-map/{serviceName}',
```

Comment:
As we talked about `/api/apm/service-map/{serviceName}` and `/api/apm/service-map/all` can be combined to a single route if `serviceName` is converted to an optional query param